File: AlgorithmsProblemSolvingPython/algoexpert/easy/minimumWaitingTime.py
def minimumWaitingTime(queries):
    queries.sort()

    sum = 0

    for index, time in enumerate(queries):
        queries_left = len(queries) - (index + 1)
        sum += time * queries_left

    return sum


if __name__ == "__main__":
    queries = [3, 2, 1, 2, 6]

    print(minimumWaitingTime(queries))

File: AlgorithmsProblemSolvingJava/src/io/programminglife/algoexpert/easy/MinimumWaitingTime.java
package io.programminglife.algoexpert.easy;

import java.util.Arrays;

public class MinimumWaitingTime {

    public static int minimumWaitingTime(int[] queries) {
        int sum = 0;

        Arrays.sort(queries);

        for (int i = 0; i < queries.length; i++) {
            int queriesLeft = queries.length - (i + 1);
            sum += queries[i] * queriesLeft;
        }

        return sum;
    }

    public static void main(String[] args) {
        int[] queries = {3, 2, 1, 2, 6};

        System.out.println(minimumWaitingTime(queries));
    }

}
